lua/nvim_lsp/mpls.lua

```diff
@@ -0,0 +1,143 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local name = "mpls"
+
+local function file_exists(name)
+   local f=io.open(name,"r")
+   if f~=nil then io.close(f) return true else return false end
+end
+
+local function get_active_interpreter()
+  local f = io.popen("which python | tr -d '\n'") -- runs command
+  local l = f:read("*a") -- read output of command
+  f:close()
+  return l
+end
+
+local function get_python_version()
+  local f = io.popen("python --version 2>&1 | awk '{print $2}' | tr -d '\n'") -- runs command
+  local l = f:read("*a") -- read output of command
+  f:close()
+  return l
+end
+
+local function make_installer()
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, name}
+
+  local bin = P{install_dir, "Microsoft.Python.LanguageServer.dll"}
+  local cmd = {"dotnet", "exec", bin}
+
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not util.has_bins("sh", "chmod", "unzip") then
+      error('Need the binaries "sh", "chmod", "unzip" to install this')
+      return
+    end
+    local script = [=[
+set -e
+# Install dotnet if not available.
+which dotnet  >/dev/null || {
+  test -x dotnet || {
+    curl -L https://dot.net/v1/dotnet-install.sh | sh
+  }
+  export PATH=$PWD:$PATH
+}
+
+# clone project
+if [ "$(uname)" == "Darwin" ]
+  then 
+    curl -fLO https://pvsc.azureedge.net/python-language-server-stable/Python-Language-Server-osx-x64.0.4.114.nupkg
+elif [ "$(uname)" == "Linux" ]
+  then
+    curl -fLO https://pvsc.azureedge.net/python-language-server-stable/Python-Language-Server-linux-x64.0.4.114.nupkg
+fi
+unzip Python-Language-Server-osx-x64.0.4.114.nupkg
```

Comment:
just unzip for osx?

Reply:
whoops! fixed. Honestly, I don't like the hardcoded curl commands anyways... I need to figure out how to replicated the xml parsing used by the emacs/vs-code versions